### PR TITLE
Fixes #6764 - removed extra/ directory

### PR DESCRIPTION
**Pull request by lzap**
http://koji.katello.org/koji/taskinfo?taskID=132151

### CODE REVIEW

**Comment by lzap** [comment]
Rebased.

http://koji.katello.org/koji/taskinfo?taskID=206052
http://koji.katello.org/koji/taskinfo?taskID=206054
http://koji.katello.org/koji/taskinfo?taskID=206074

**Comment by domcleal** [comment]
open4/ftools are still dependencies of foreman_discovery 1.4.1, see https://github.com/theforeman/foreman-packaging/pull/288#issuecomment-51762353, so it'll probably cause runtime errors.

**Comment by lzap** [comment]
Right, we removed it from develop branch only. Let's merge this after we release our first RC next week. Thanks. https://rubygems.org/gems/foreman_discovery

**Comment by lzap** [comment]
I have added a commit with 2.0.0.rc1 update.

File: rubygem-foreman_discovery/rubygem-foreman_discovery.spec [hunk 1]
```diff
@@ -34,12 +34,8 @@ License:    GPLv3
...
 Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}%{?prever}.gem
 
-Requires:   foreman >= 1.6.0
-Requires:   %{?scl_prefix}rubygem(open4)
-Requires:   %{?scl_prefix}rubygem(ftools)
-Requires:   advancecomp
-Requires:   squashfs-tools
-Requires:   sudo
+Requires:   foreman >= 1.7.0
+Requires:   %{?scl_prefix}rubygem(deface)
```
Comment (domcleal) [inline rubygem-foreman_discovery/rubygem-foreman_discovery.spec]: < 1.0

File: rubygem-foreman_discovery/rubygem-foreman_discovery.spec [hunk 2]
```diff
@@ -92,28 +88,17 @@ cat <<GEMFILE > %{buildroot}%{foreman_bundlerd_dir}/%{gem_name}.rb
...
-# Output directory of the imgae build task
-mkdir -p %{buildroot}/%{foreman_dir}/discovery_image
-
 %files
 %dir %{gem_instdir}
 %{gem_instdir}/app
 %{gem_instdir}/lib
 %{gem_instdir}/config
-%{gem_instdir}/extra
+%{gem_instdir}/db
```
Comment (domcleal) [inline rubygem-foreman_discovery/rubygem-foreman_discovery.spec]: also add %posttrans section with db:migrate, db:seed, apipie:cache.  Copy: https://github.com/theforeman/foreman-packaging/blob/rpm/develop/rubygem-foreman_bootdisk/rubygem-foreman_bootdisk.spec#L104-L110

**Comment by lzap** [comment]
Thanks for the quick review, updated.

**Comment by domcleal** [comment]
Is this to be released into nightly repos only, or 1.7 too?

**Comment by domcleal** [comment]
Oh, and lastly, could you delete ftools/open4 directories and from rel-eng/tito.props?

**Comment by lzap** [comment]
Delete `ftools` as well? Yes both 1.7 and nightly.

**Comment by domcleal** [comment]
I can't see any other use of the gems - if you're removing from comps, we should remove the source dirs.

**Comment by lzap** [comment]
Removed ftools as well. Pushed.

**Comment by domcleal** [comment]
The directories are still there?

File: rubygem-foreman_discovery/rubygem-foreman_discovery.spec [hunk 3]
```diff
@@ -34,12 +34,8 @@ License:    GPLv3
 URL:        http://github.com/theforeman/foreman_discovery
 Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}%{?prever}.gem
 
-Requires:   foreman >= 1.6.0
-Requires:   %{?scl_prefix}rubygem(open4)
-Requires:   %{?scl_prefix}rubygem(ftools)
-Requires:   advancecomp
-Requires:   squashfs-tools
-Requires:   sudo
+Requires:   foreman >= 1.7.0
```
Comment (domcleal) [inline rubygem-foreman_discovery/rubygem-foreman_discovery.spec]: 1.6.0, according to engine.rb

Comment (lzap) [inline rubygem-foreman_discovery/rubygem-foreman_discovery.spec]: This is a bug, we require 1.7+. Filed a PR for RC2: https://github.com/theforeman/foreman_discovery/pull/128

**Comment by lzap** [comment]
Squashed the removal of the dirs into the first commit.

**Comment by domcleal** [comment]
Thanks @lzap, building for nightly and 1.7.

**Pull request merged**